src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -61,22 +61,51 @@ public void storeMessage(Message message) {
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    for (Entity entity : results.asIterable()) {
-      try {
-        String idString = entity.getKey().getName();
-        UUID id = UUID.fromString(idString);
-        String text = (String) entity.getProperty("text");
-        long timestamp = (long) entity.getProperty("timestamp");
-
-        Message message = new Message(id, user, text, timestamp);
-        messages.add(message);
-      } catch (Exception e) {
-        System.err.println("Error reading message.");
-        System.err.println(entity.toString());
-        e.printStackTrace();
-      }
-    }
-
-    return messages;
+    return getMessagesHelper(results); 
   }
+
+   /**
+   * Gets messages posted by all users.
+   *
+   * @return a list of messages posted by all users, or empty list if there are no
+   * messages posted. List is sorted by time descending.
+   */
+  public List<Message> getAllMessages(){
+  List<Message> messages = new ArrayList<>();
+
+  Query query = new Query("Message")
+    .addSort("timestamp", SortDirection.DESCENDING);
+  PreparedQuery results = datastore.prepare(query);
+
+  return getMessagesHelper(results); 
+ }
+
+ /**
+   * Helper function for getMessages and getAllMessages. 
+   *
+   * @return a list of messages, or empty list if there are no
+   * messages posted. List is sorted by time descending.
```

Comment:
One minor point: this list isn't _necessarily_ sorted by time descending--that's actually controlled by the queries on lines 66 and 71. We could reuse this helper method in the future with queries that sort by something else, if we wanted to, so we don't need to say the last sentence in this comment.